### ISSUE TITLE
using it from a web site?

### ISSUE THREAD

**Issue by lonnietc** [pedroetb/tts-api#1]
Hello,

I was just came across your tts-api on Docker Hub that I tested out on my local system.

I found that the curl calls do not seem to work on my Debian 10 system and that may just be the device setting that I can test out later.

The question that I have is this. 

I was able to start up the container and use my web browser for a local interface:

http://localhost:3000

I would like to be able to use this on a website on a web server that I have set up and need to see how to interface it the same way that you do for your local test page so that I can send text to it in the background.

Can you please advise on how I might be able to do this?
Thanks in advance

**Comment by pedroetb** [comment]
Hi,

where you want to listen the speech? You should run the container on that machine.

Later, you can use a reverse proxy to expose it to the internet, and should be able to use curl against it. Maybe you had some system/container network misconfiguration in previous tests?

**Comment by lonnietc** [comment]
Hello,

Thanks for getting back to me.

I was able to get things to work locally but for the goal that I am trying to do, I have a website. Actually, it is a small IPA (Internet Personal Assistant) written mostly in Python and want to add sound so that, in much the same way as your localhost:3000 will pay back a text input in the browser, I want to be able to send my IPA text responses and have it speak back to the user that is on the website.

I am probably making this harder than it has to be, but I am guessing that I just need to look through some code to see how it is done.

Basically, I am just trying to integrate the use of your container into a existing website that will speak to the users.

Cheers,

**Comment by lonnietc** [comment]
Hello,

I have once more question on your TTS server.

I want to run it on a server and integrated into my web pages that will have multiple on the web site at the same time. 

Will this server be able to handle multiple independent requests from the web server for the different users?

Also, I am investigating the possible use of a Melgan-Tacotron2 TTS model that could possible be added to your selections from GoogleSpeech, gTTS, Festival, and eSpeak, but want to see if the current approach can handle multiple simultaneous queries to the TTS-API docker container.

Cheers

Thanks again

**Comment by lonnietc** [comment]
Hello Again,

I was just reviewing your TTA-API server software and wanted to ask about something.

I plan to try and utilize your TTS-API serve an external docker container running on a server system but need to be able to run it in such a way that it does not need the "--device /dev/snd".

Also, since the responses from TTS-API are to be used on a website, it would be best if the API call could return a "wav", or "mp3" file, perhaps.

I was thinking that maybe a call like this:

$ curl http://someserver.com:3000 \
  -d '{ "voice": "google_speech", "textToSpeech": "hello world", "language": "en", "speed": "1" }' \
  -H "Content-Type: application/json"

could return something like "results.wav" or "results.mp3"

Then that could be played locally, or sent to the client user web browser from the web server that would be played locally.

Do you have any ideas on how I might be able to change your current TTS-API server such that it does not need the /dev/snd device and can return a wav audio file?

Any help that you could provide wold be greatly appreciated.
Thanks

**Comment by pedroetb** [comment]
Ok, your usage case is not covered by this project. It is focused on playing directly on local sound device.
But I think it would be possible to response with an audio file, as you suggest.

With local sound, you can only play 1 speech at the same time, but generating audio files you can serve multiple speech concurrently.
Some changes may be necessary to achieve this, both API REST and speech commands, but it would be nice to implement this as an optional behaviour!

To integrate new speech engines, we need isolate dependencies and clarify shell commands to run (and update api configuration to accept a new option).

I will check this soon, thanks!

**Comment by lonnietc** [comment]
Thanks for getting back to me and I look forward to hearing from you.

Yea, since this will be serving users on a website, it will need to handle
many multiple inquiries, perhaps some simultaneously, and it seems that
serving a file back would be the best approach.

For the new speech engine, I can get in touch with you offline from the
github to see if you like the new engine that I have in mind to see if it
can be added, ok.
Cheers,

**Comment by pedroetb** [comment]
Initial audio download support added at v2.5.0
